hack/generator/pkg/astmodel/armconversion/convert_to_arm_function_builder.go

```diff
@@ -498,3 +537,11 @@ func callToArmFunction(source ast.Expr, destination ast.Expr, methodName string)
 
 	return results
 }
+
+func typeRequiresCopying(theType astmodel.Type) bool {
+	switch theType.(type) {
+	case *astmodel.OptionalType, *astmodel.MapType, *astmodel.ArrayType:
```

Comment:
Why does an `OptionalType` require copying? Doesn't that depend on what's wrapped?

Reply:
No, because it's implemented as a pointer. If the conversion function just has:
```go
k8sResource.OptionalString = armResource.OptionalString
```
... and then some code assigns through the pointer:
```go
*armResource.OptionalString = "oops"
```
... then `k8sResource` will see that update.

https://play.golang.org/p/RSR5io24eXB

Reply:
I mean, it does depend on what's wrapped as well. If it's a generated non-enum type then it won't come through this check because the source and destination types will be different - one will be an Arm type.

Reply:
Fair enough.